cpp/src/groupby/sort/sort_helper.cu

```diff
@@ -88,6 +89,28 @@ namespace cudf {
 namespace groupby {
 namespace detail {
 namespace sort {
+
+sort_groupby_helper::sort_groupby_helper(table_view const& keys,
+                                         null_policy include_null_keys,
+                                         sorted keys_pre_sorted)
+  : _unflattened_keys(keys),
+    _num_keys(-1),
+    _keys_pre_sorted(keys_pre_sorted),
+    _include_null_keys(include_null_keys)
+{
+  using namespace cudf::structs::detail;
+
+  auto [flattened_keys, _, __, struct_null_vectors] =
+    flatten_nested_columns(keys, {}, {}, column_nullability::FORCE);
+  _struct_null_vectors = std::move(struct_null_vectors);
+  _keys                = flattened_keys;
+
+  if (keys_pre_sorted == sorted::YES and include_null_keys == null_policy::EXCLUDE and
```

Comment:
A comment here would be helpful

Reply:
Hmm. I agree. (As an aside, this is pre-existing code moved out of `sort_helper.hpp`.)

My understanding is that this is to enable re-sorting the input even if the caller indicates that the input is pre-sorted. We do this so as to force null keys to the end of the output column, for easier filtering.

@devavret will keep me honest here.